### PR TITLE
Feature/assignment 03

### PR DESCRIPTION
**Pull request by ayemoekyaw99**

### CODE REVIEW

File: assignment-03/app/Dao/StudentDao.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        return Student::select('students.*')
+                            ->leftJoin('majors','students.major_id','majors.id')
+                            ->when(request('searchKey'),function($query){
+                            $key=request('searchKey');
+                            $query->where('students.name', 'like', '%' . $key . '%')
+                            ->orWhere('students.email', '=', $key)
+                            ->orWhere('students.phone', '=', $key)
+                            ->orWhere('students.address', 'like', '%' . $key . '%')
+                            ->orWhere('majors.name', 'like', '%' . $key . '%');
+                            })->get();
```
Comment (ingyinmaycgm) [inline assignment-03/app/Dao/StudentDao.php:27]: - Please make vertically align `:: & ->`
- Fix query indentation.
- Search major name with where = . Not where like.
- How about raw query?

**Review by thinpyantsoe — APPROVED** [review]
1) php artisan migrate error
  majors table after studens table create
  General error: 1824 Failed to open the referenced table 'majors' (Connection: mysql, SQL: 
  alter table `students` add constraint `students_major_id_foreign` foreign key

2) major edit.blade where back button click this error show
   App\Http\Controllers\MajorController::edit(): Argument 
   https://github.com/ayemoekyaw99/PHP_Training_Test/pull/1 ($id) must be of type int,
    string given
3) no choose file if import button click this error show
   No ReaderType or WriterType could be detected. Make sure you either pass a valid 
   extension to the filename or pass an explicit type.

4) pls format in you create all file

File: assignment-03/app/Dao/MajorDao.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  * Major update
+
+  * @param mixed $data
+  * @return void
+  */
+    public function update(array $data, int $id): void
+    {
+        $major = Major::find($id);
+        $major->update($data);
+
```
Comment (thinpyantsoe) [inline assignment-03/app/Dao/MajorDao.php:62]: Remove empty line.

File: assignment-03/app/Dao/StudentDao.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+     * @return string
+     */
+    public function find($id)
+    {
+        return Student::find($id);
+    }
+
+    /**
+   * Student update
+
```
Comment (thinpyantsoe) [inline assignment-03/app/Dao/StudentDao.php:73]: pls format document.

File: assignment-03/app/Dao/StudentDao.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+   * @param mixed $data
+   * @return void
+   */
+    public function update(array $data, int $id): void
+    {
+        $student = Student::find($id);
+        $student->update($data);
+
```
Comment (thinpyantsoe) [inline assignment-03/app/Dao/StudentDao.php:81]: Remove Empty line.

File: assignment-03/app/Http/Controllers/ExcelController.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+class ExcelController extends Controller
+{
+
+
+    /**
+     * @return \Illuminate\Support\Collection
+     */
+    // public function import()
+    // {
```
Comment (thinpyantsoe) [inline assignment-03/app/Http/Controllers/ExcelController.php:20]: Delete this if no use

File: assignment-03/app/Services/StudentService.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        return $this->studentDao->find($id);
+
+    }
+
+    /**
+     * sample function
+     *
+     * @return string
+     */
+
```
Comment (thinpyantsoe) [inline assignment-03/app/Services/StudentService.php:80]: Remove this line

File: assignment-03/resources/views/layout/app.blade.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>@yield('title')</title>
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.2/dist/css/bootstrap.min.css" integrity="sha384-xOolHFLEh07PJGoPkLv1IbcEPTNtaed2xpHsD9ESMhqIYd0nLMwNLD69Npy4HI+N" crossorigin="anonymous" />
+</head>
+
+<body>
+    <nav class="navbar navbar-expand-lg navbar-light bg-light">
+        <div class="container">
+            {{--<a class="navbar-brand" href="{{route('student#list')}}">Navbar</a>--}}
```
Comment (thinpyantsoe) [inline assignment-03/resources/views/layout/app.blade.php:15]: Remove no use line.

File: assignment-03/resources/views/layout/app.blade.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+                <span class="navbar-toggler-icon"></span>
+            </button>
+
+            <div class="collapse navbar-collapse" id="navbarSupportedContent">
+                <ul class="navbar-nav ml-auto">
+                    <li class="nav-item active">
+                        <a class="nav-link" href="{{route('students#list')}}">Students</a>
+                    </li>
+                    <li class="nav-item">
+                        <a class="nav-link" href="{{route('majors#list')}}">Majors</a>
```
Comment (thinpyantsoe) [inline assignment-03/resources/views/layout/app.blade.php:26]: nav major no active if click majors.

File: assignment-03/resources/views/major/create.blade.php [hunk 9]
```diff
@@ -0,0 +1,39 @@
+@extends('layout.app')
+
+@section('Major Create')
+@endsection
+
+@section('content')
+<div class="container">
+    <div class=" mt-3">
+        @if (session('success'))
```
Comment (thinpyantsoe) [inline assignment-03/resources/views/major/create.blade.php:9]: pls blade format.

File: assignment-03/resources/views/student/list.blade.php [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach ($students as $student)
+                    <tr>
+                        <td>{{ $student->id }}</td>
+                        <td>{{ $student->name }}</td>
+                        @foreach ($majors as $major)
+                        @if ($student->major_id == $major->id)
+                        <td>
```
Comment (thinpyantsoe) [inline assignment-03/resources/views/student/list.blade.php:62]: pls format document

**Review by thinpyantsoe — APPROVED** [review]
pls format in you create all file

File: assignment-03/app/Contracts/Services/StudentServiceInterface.php [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Contracts\Services;
+
+interface StudentServiceInterface
+{
+    public function getAllStudents();
+    public function getAllMajors();
```
Comment (thinpyantsoe) [inline assignment-03/app/Contracts/Services/StudentServiceInterface.php:8]: If you break with empty line, then break all functions with empty line.
If your don't use new empty line, then remove new lines.
Make all the same format.

File: assignment-03/app/Contracts/Services/StudentServiceInterface.php [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    public function create(array $data);
+
+    public function destroy(int $id);
+
+    public function getStudentById(int $id);
+
+    public function update(array $data, int $id);
+}
```
Comment (thinpyantsoe) [inline assignment-03/app/Contracts/Services/StudentServiceInterface.php:16]: Add eof.

File: assignment-03/app/Dao/MajorDao.php [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    /**
+  * Major update
+
+  * @param mixed $data
+  * @return void
+  */
+    public function update(array $data, int $id): void
+    {
+        $major = Major::find($id);
+        $major->update($data);
```
Comment (thinpyantsoe) [inline assignment-03/app/Dao/MajorDao.php:61]: Use only update query. Major::where()->update.
Print Query Log and check the difference.
\DB::enableQueryLog()
//DB statement
dd(\DB::getQueryLog());

**Review by thinpyantsoe — APPROVED** [review]

**Pull request merged**